feature/bgp/routereflector/route-reflector-test

```diff
@@ -0,0 +1,45 @@
+Summary: 
+ - BGP route reflector capabilities check
+ - Ensure functionality of different OC paths for "supported-capabilities", "BGP peer-type", "BGP Neighbor details" and "BGP transport session parameters"
+
+Topology:
+ATE (Port1) <-EBGP-> (Port1) DUT (Port2) <-IS-IS/IBGP-> (Port2) ATE, DUT (Port3) <-IS-IS/IBGP-> (Port3) ATE
```

Comment:
Not review feedback, really, but rather an observation: it would be useful to have variations of this test that don't have dependencies on IS-IS etc. Regardless I think this test is a realistic case, so this test makes sense to me.

Reply:
This is more to capture our use case. IMO, we can add more at a later date